admin-dev/themes/new-theme/js/pages/tax-rules-group/TaxRulesGroupForm.js

```diff
@@ -0,0 +1,346 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import taxRulesGroupFormMap from './tax-rules-groups-form-map';
+
+/**
+ * Class responsible for javascript actions Tax rules edit form.
+ */
+export default class TaxRulesGroupForm {
+  constructor() {
+    this.taxRulesFormAction = $(taxRulesGroupFormMap.taxRuleForm).attr('action');
+    this._initEvents();
+
+    return {};
+  }
+
+  /**
+   * Initialize page's events.
+   *
+   * @private
+   */
+  _initEvents() {
+    this._initTaxRuleForm();
+    $(taxRulesGroupFormMap.countrySelect).on('change', () => this._handleCountryChange());
+    $(taxRulesGroupFormMap.editTaxRuleLink).on('click', (event) => this._handleEditTaxRuleClick(event));
+    $(taxRulesGroupFormMap.addTaxRuleBtn).on('click', (event) => this._handleAddTaxRuleClick(event));
+  }
+
+  /**
+   * Checks if form has non empty action witch indicates form was submitted before
+   * to initiate form before user interaction. Also check if list is empty and open create form
+   * if it is
```

Comment:
Hmmm, I don't know if this comment is right. Docblock should say what functions does, but should not go into details of how. Here you described every `if` statement inside, maybe it's too much. :thinking: 

@matks wdyt?

Reply:
> Docblock should say what functions does, but should not go into details of how.

I disagree 😉 docblock should provide additional explanations that are valuable. Most of the time it's about the inputs and the outputs, sometimes also the intent and what it does, but when the "how" is complex enough we can put it here as well. What matters is to make it valuable, if the code is simple enough we dont need to explain it.

Reply:
Although there is true to that, I could argue about it, especially in this case. I don't think it makes sense to document `if` statements. Code is not that complex, so instead of having this (see below code) with comments explaining what it does

```javascript
 if ($(taxRulesGroupFormMap.taxRulesGrid).find('.grid-table-empty').length) {
      this._showTaxRulesForm($(taxRulesGroupFormMap.addLink).attr('href'));
}
```
I'd rather see this:

```javascript
const isListEmpty = ...;

 if (isListEmpty) {
      this._openTaxRulesForm();
}
```